cmd/zc_traverser_blob.go

```diff
@@ -95,15 +97,21 @@ func (t *blobTraverser) IsDirectory(isSource bool) bool {
 			msg := fmt.Sprintf("Failed to check if the destination is a folder or a file (Azure Files). Assuming the destination is a file: %s", err)
 			azcopyScanningLogger.Log(pipeline.LogError, msg)
 		}
-		return false
+		return false, nil
 	}
 
 	if len(resp.Segment.BlobItems) == 0 {
 		//Not a directory
-		return false
+		if stgErr, ok := blobErr.(azblob.StorageError); ok {
```

Comment:
Is this a valid error for list blobs to throw? I thought it returns no error if file doesnt exist

Reply:
that's why we're checking the blobErr, if you scroll up you'll see that this is retrieved on the getProps call. If nothing was listed then we check if blobErr had a not exist - this would confirm that 1. the hdi file doesn't exist AND nothing with this dir prefix exists so throw.